src/triage/component/architect/builders.py

```diff
@@ -247,6 +250,8 @@ def build_matrix(
             self.db_config["cohort_table_name"], self.db_engine
         ):
             logging.warning("cohort table is not populated, cannot build matrix")
+            if self.run_id:
+                errored_matrix(self.run_id, self.db_engine)
```

Comment:
(This could pass `self.sessionmaker` ... which itself could be a `cachedproperty`.)